lager/lenses.hpp

```diff
@@ -123,51 +128,103 @@ auto attr(Member member)
     });
 }
 
+/*!
+ * `Key -> Lens<{X}, [X]>`
+ */
 template <typename Key>
-auto at(Key key)
-{
-    return zug::comp([=](auto&& f) {
-        return [f, &key](auto&& p) {
-            return f([&] {
+auto at(Key key) {
+    return zug::comp([key](auto&& f) {
+        return [f, &key](auto&& whole) {
+            using Part = std::optional<std::decay_t<decltype(whole.at(key))>>;
+
+            return f([&]() -> Part {
                 try {
-                    return std::forward<decltype(p)>(p).at(key);
-                } catch (std::out_of_range const&) {
-                    return std::decay_t<decltype(p.at(key))>{};
+                    return std::forward<decltype(whole)>(whole).at(key);
+                } catch (std::out_of_range const&) { return std::nullopt; }
+            }())([&](Part part) {
```

Comment:
A copy is the easy way out here. I tried really hard to avoid constructing optionals when passed a value but it ended up being way [overkill](https://gist.github.com/Siapran/6e6a87ac5043e13a89bbb289dad4aacf)...

Reply:
Still not sure if this is the best way.